app/services/user_silencer.rb

```diff
@@ -57,8 +57,11 @@ def silence
           user_history: @user_history,
           post_id: @opts[:post_id],
           silenced_till: @user.silenced_till,
-          silenced_at: DateTime.now
+          silenced_at: DateTime.now,
+          silence_message_params: silence_message_params
         )
+
+        SystemMessage.create(@user, message_type, **silence_message_params)
```

Comment:
Why do you need the splat operator here?

Reply:
Ruby 2.7.1 deferred this (no longer warns), but there was previous a bunch of warnings in 2.7.0 about how passing an options hash as a variable (and having it auto-convert to keyword args) was deprecated: https://blog.bigbinary.com/2020/04/14/ruby-2-7-deprecates-conversion-of-keyword-arguments.html

The splat operator makes that conversion explicit.

Reply:
Wait, you're right, there's no reason for it - I was misreading the function definition, and thought the args hash argument was a keyword hash.